src/query/graphite/native/builtin_functions.go

```diff
@@ -94,6 +95,46 @@ func sortByMaxima(ctx *common.Context, series singlePathSpec) (ts.SeriesList, er
 	return highestMax(ctx, series, len(series.Values))
 }
 
+/*
+Compares the maximum of each series against the given `value`. If the series
+maximum is greater than `value`, the regular expression search and replace is
+applied against the series name to plot a related metric
+
+e.g. given useSeriesAbove(ganglia.metric1.reqs,10,'reqs','time'),
+the response time metric will be plotted only when the maximum value of the
+corresponding request/s metric is > 10
+
+&target=useSeriesAbove(ganglia.metric1.reqs,10,"reqs","time")
+*/
+func useSeriesAbove(ctx *common.Context, seriesList singlePathSpec , maxAllowedValue float64, search, replace string) (ts.SeriesList, error) {
+	results := make([]*ts.Series, 0, len(seriesList.Values))
+	for _, series := range seriesList.Values {
+		if series.SafeMax() > maxAllowedValue {
+			opts := storage.FetchOptions{
+				StartTime: ctx.StartTime,
+				EndTime:   ctx.EndTime,
+				DataOptions: storage.DataOptions{
+					Timeout: ctx.Timeout,
+				},
+			}
+			seriesName := strings.Replace(series.Name(), search, replace, 1)
```

Comment:
Should this ignore series that don’t have `search` in them? Also this may need to replace a specific metric part, although if this is how its done in graphite thats probably fine

Reply:
Actually looking at [graphite's implementation](https://github.com/graphite-project/graphite-web/blob/415e571a1b2276395419c9a34c8c492d52a804df/webapp/graphite/render/functions.py#L3520) this should be `strings.Replace(series.Name(), search, replace, -1)` to ensure each string is replaced; we can probably do one better and check to see if any replacements happened, and if they didn't, return the original series

Reply:
I updated this to `-1` and copied the logic from the `graphite-web` implementation, so we return the original series if there is no replacement.